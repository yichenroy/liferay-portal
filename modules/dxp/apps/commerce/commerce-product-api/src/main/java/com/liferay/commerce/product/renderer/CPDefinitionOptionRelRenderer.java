/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}