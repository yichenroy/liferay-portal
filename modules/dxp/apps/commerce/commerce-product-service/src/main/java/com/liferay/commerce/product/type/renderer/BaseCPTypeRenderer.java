/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}