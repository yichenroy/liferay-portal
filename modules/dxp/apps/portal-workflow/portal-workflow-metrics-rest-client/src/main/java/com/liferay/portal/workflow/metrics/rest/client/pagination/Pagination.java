/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 *
 *
 *
 */

package com.liferay.portal.workflow.metrics.rest.client.pagination;

import javax.annotation.Generated;

/**
 * @author Rafael Praxedes
 * @generated
 */
@Generated("")
public class Pagination {

	public static Pagination of(int page, int pageSize) {
		return new Pagination(page, pageSize);
	}

	public int getEndPosition() {
		return _page * _pageSize;
	}

	public int getPage() {
		return _page;
	}

	public int getPageSize() {
		return _pageSize;
	}

	public int getStartPosition() {
		return (_page - 1) * _pageSize;
	}

	private Pagination(int page, int pageSize) {
		_page = page;
		_pageSize = pageSize;
	}

	private final int _page;
	private final int _pageSize;

}