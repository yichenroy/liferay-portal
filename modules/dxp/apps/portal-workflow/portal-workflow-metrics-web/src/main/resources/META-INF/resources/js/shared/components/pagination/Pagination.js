import PageItem from './PageItem';
import PageLink from './PageLink';
import React from 'react';
import { withRouter } from 'react-router-dom';

/**
 * @class
 * @memberof shared/components
 */
class Pagination extends React.Component {
	constructor(props) {
		super(props);
		this.state = {};
	}

	getProps() {
		const {
			match: { params },
			maxPages,
			totalCount
		} = this.props;
		const { page = 1, pageSize = 20 } = params || {};

		return {
			maxPages,
			page: Number(page),
			pageSize: Number(pageSize),
			totalCount
		};
	}

	buildDropMenu(initCount, condition, functionIterator) {
		const data = [];

		for (let i = initCount; i < condition; i++) {
			if (functionIterator) {
				functionIterator(data, i);
			}
			else {
				data.push(i);
			}
		}

		return (
			<li className="dropdown page-item" key={`process_list_pag_${initCount}`}>
				<a
					className="dropdown-toggle page-link"
					data-senna-off
					data-toggle="dropdown"
					href="javascript:;"
				>
					<span aria-hidden="true">{'...'}</span>

					<span className="sr-only" />
				</a>
				<div className="dropdown-menu dropdown-menu-top-center">
					<ul
						className="inline-scroller link-list"
						data-current-index={`${initCount}`}
						key={`process_list_pag_sublist_${initCount}`}
					>
						{data.map(page => (
							<PageLink
								key={`process_list_pag_sublist_item_${page}`}
								page={page}
							/>
						))}
					</ul>
				</div>
			</li>
		);
	}

	render() {
		const {
			maxPages,
			page: activePage,
			pageSize,
			totalCount
		} = this.getProps();

		const lastPage = Math.ceil(totalCount / pageSize);

		const nextPage = activePage + 1;
		const prevPage = activePage - 1;

		const hasNextPage = nextPage <= lastPage;
		const hasPreviousPage = prevPage > 0;

		const pages = lastPage;

		const renderPageItems = () => {
			const rows = {};

			const wasAdded = page => Object.keys(rows).indexOf(page) > -1;

			const addPage = (page, component) => {
				if (page > 0 && !wasAdded(page)) {
					rows[page] = component;
				}
			};

			if (pages <= 5) {
				for (let i = 1; i <= pages; i++) {
					addPage(
						i,
						<PageItem
							highlighted={i === activePage}
							key={`process_list_pag_${i}`}
							page={i}
						/>
					);
				}
			}
			else if (activePage === 1) {
				for (let i = 1; i <= 3; i++) {
					addPage(
						i,
						<PageItem
							highlighted={i === activePage ? true : false}
							key={`process_list_pag_sublist_${i}`}
							page={i}
						/>
					);
				}

				addPage(
					4,
					this.buildDropMenu(4, maxPages, (data, i) => {
						if (i < pages) {
							data.push(i);
						}
					})
				);

				addPage(
					lastPage,
					<PageItem
						highlighted={lastPage === activePage ? true : false}
						key={`process_list_pag_${lastPage}`}
						page={lastPage}
					/>
				);
			}
			else if (activePage === pages) {
				addPage(
					1,
					<PageItem
						highlighted={1 === activePage ? true : false}
						key={'process_list_pag_1'}
						page={1}
					/>
				);

				addPage(
					2,
					this.buildDropMenu(
						2,
						maxPages > activePage - 2 ? activePage - 2 : maxPages
					)
				);

				for (let i = pages - 2; i <= pages; i++) {
					addPage(
						i,
						<PageItem
							highlighted={i === activePage ? true : false}
							key={`process_list_pag_sublist_${i}`}
							page={i}
						/>
					);
				}
			}
			else {
				addPage(
					1,
					<PageItem
						highlighted={activePage === 1 ? true : false}
						key={'process_list_pag_1'}
						page={1}
					/>
				);
				if (activePage - 3 > 1) {
					addPage(
						2,
						this.buildDropMenu(
							2,
							maxPages > activePage - 1 ? activePage - 1 : maxPages
						)
					);
				}
				else {
					for (
						let i = 2,
							x = maxPages > activePage - 1 ? activePage - 1 : maxPages;
						i < x;
						i++
					) {
						addPage(
							i,
							<PageItem
								highlighted={i === activePage ? true : false}
								key={`process_list_pag_sublist_${i}`}
								page={i}
							/>
						);
					}
				}

				if (activePage - 1 > 1) {
					addPage(
						activePage - 1,
						<PageItem
							key={`process_list_pag_sublist_${activePage - 1}`}
							page={activePage - 1}
						/>
					);
				}

				addPage(
					activePage,
					<PageItem
						highlighted={activePage !== 1 ? true : false}
						key={`process_list_pag_active_${activePage}`}
						page={activePage}
					/>
				);

				if (activePage + 1 < pages) {
					addPage(
						activePage + 1,
						<PageItem
							key={`process_list_pag_plus_${activePage + 1}`}
							page={activePage + 1}
						/>
					);
				}
				const remainingPages =
					pages - activePage + 2 < maxPages ? pages : activePage + 2 + maxPages;

				if (activePage + 3 < pages) {
					addPage(
						activePage + 2,
						this.buildDropMenu(activePage + 2, remainingPages)
					);
				}
				else {
					for (let i = activePage + 2; i < remainingPages; i++) {
						addPage(
							i,
							<PageItem
								highlighted={i === activePage}
								key={`process_list_pag_sublist_${i}`}
								page={i}
							/>
						);
					}
				}

				addPage(
					lastPage,
					<PageItem
						highlighted={lastPage === activePage}
						key={`process_list_pag_last_${lastPage}`}
						page={lastPage}
					/>
				);
			}

			return Object.keys(rows).map(key => rows[key]);
		};

		return (
			<ul className="pagination pull-right">
				<PageItem
					disabled={!hasPreviousPage}
					key={`process_list_pag_prev_${prevPage}`}
					page={hasPreviousPage ? prevPage : 1}
					type="prev"
				/>

				{renderPageItems()}

				<PageItem
					disabled={!hasNextPage}
					key={`process_list_pag_next_${nextPage}`}
					page={nextPage}
					type="next"
				/>
			</ul>
		);
	}
}

export default withRouter(Pagination);

export { Pagination };