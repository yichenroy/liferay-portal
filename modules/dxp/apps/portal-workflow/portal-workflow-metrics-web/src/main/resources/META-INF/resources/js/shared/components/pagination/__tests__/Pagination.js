import Pagination from '../Pagination';
import React from 'react';
import renderer from 'react-test-renderer';

test('Should render component', () => {
	const onSelectPage = () => page => page;

	const component = renderer.create(
		<Pagination onSelectPage={onSelectPage()} pageSize={1} totalCount={10} />
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should change page', () => {
	const onSelectPage = () => page => page;

	const component = shallow(
		<Pagination onSelectPage={onSelectPage()} pageSize={1} totalCount={10} />
	);

	const instance = component.instance();

	instance.goToPage(2);
	expect(component).toMatchSnapshot();
});

test('Should render component at page 1', () => {
	const onSelectPage = () => page => page;

	const component = renderer.create(
		<Pagination
			onSelectPage={onSelectPage()}
			page={1}
			pageSize={5}
			totalCount={26}
		/>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should render component at page 2', () => {
	const onSelectPage = () => page => page;

	const component = renderer.create(
		<Pagination
			onSelectPage={onSelectPage()}
			page={2}
			pageSize={5}
			totalCount={26}
		/>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should render component at page 3', () => {
	const onSelectPage = () => page => page;

	const component = renderer.create(
		<Pagination
			onSelectPage={onSelectPage()}
			page={3}
			pageSize={5}
			totalCount={26}
		/>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should render component at page 4', () => {
	const onSelectPage = () => page => page;

	const component = renderer.create(
		<Pagination
			onSelectPage={onSelectPage()}
			page={4}
			pageSize={5}
			totalCount={26}
		/>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should render component at page 5', () => {
	const onSelectPage = () => page => page;

	const component = renderer.create(
		<Pagination
			onSelectPage={onSelectPage()}
			page={5}
			pageSize={5}
			totalCount={26}
		/>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});