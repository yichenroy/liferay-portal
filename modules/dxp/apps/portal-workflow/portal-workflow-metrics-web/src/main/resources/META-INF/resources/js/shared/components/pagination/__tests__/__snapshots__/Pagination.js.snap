// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should change page 1`] = `
<ul
  className="pagination pull-right"
>
  <PageItem
    disabled={false}
    key="process_list_pag_prev_1"
    onChangePage={[Function]}
    page={1}
    type="prev"
  />
  <PageItem
    highlighted={false}
    key="process_list_pag_1"
    onChangePage={[Function]}
    page={1}
  />
  <PageItem
    highlighted={true}
    key="process_list_pag_active_2"
    onChangePage={[Function]}
    page={2}
  />
  <PageItem
    key="process_list_pag_plus_3"
    onChangePage={[Function]}
    page={3}
  />
  <li
    className="dropdown page-item"
    key="process_list_pag_4"
  >
    <a
      className="dropdown-toggle page-link"
      data-toggle="dropdown"
      href="javascript:;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
      <span
        className="sr-only"
      />
    </a>
    <div
      className="dropdown-menu dropdown-menu-top-center"
    >
      <ul
        className="inline-scroller link-list"
        data-current-index="4"
        key="process_list_pag_sublist_4"
      >
        <PageLink
          key="process_list_pag_sublist_item_4"
          onChangePage={[Function]}
          page={4}
        />
        <PageLink
          key="process_list_pag_sublist_item_5"
          onChangePage={[Function]}
          page={5}
        />
        <PageLink
          key="process_list_pag_sublist_item_6"
          onChangePage={[Function]}
          page={6}
        />
        <PageLink
          key="process_list_pag_sublist_item_7"
          onChangePage={[Function]}
          page={7}
        />
        <PageLink
          key="process_list_pag_sublist_item_8"
          onChangePage={[Function]}
          page={8}
        />
        <PageLink
          key="process_list_pag_sublist_item_9"
          onChangePage={[Function]}
          page={9}
        />
      </ul>
    </div>
  </li>
  <PageItem
    highlighted={false}
    key="process_list_pag_last_10"
    onChangePage={[Function]}
    page={10}
  />
  <PageItem
    disabled={false}
    key="process_list_pag_next_3"
    onChangePage={[Function]}
    page={3}
    type="next"
  />
</ul>
`;

exports[`Should render component 1`] = `
<ul
  className="pagination pull-right"
>
  <li
    className="page-item disabled"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#-1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-left"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-left"
        />
      </svg>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="page-item active"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#0"
    >
      0
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="dropdown page-item"
  >
    <a
      className="dropdown-toggle page-link"
      data-toggle="dropdown"
      href="javascript:;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
      <span
        className="sr-only"
      />
    </a>
    <div
      className="dropdown-menu dropdown-menu-top-center"
    >
      <ul
        className="inline-scroller link-list"
        data-current-index="2"
      >
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#2"
          >
            <span
              className="sr-only"
            />
            2
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#3"
          >
            <span
              className="sr-only"
            />
            3
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#4"
          >
            <span
              className="sr-only"
            />
            4
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#5"
          >
            <span
              className="sr-only"
            />
            5
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#6"
          >
            <span
              className="sr-only"
            />
            6
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#7"
          >
            <span
              className="sr-only"
            />
            7
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#8"
          >
            <span
              className="sr-only"
            />
            8
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#9"
          >
            <span
              className="sr-only"
            />
            9
          </a>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#10"
    >
      10
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-right"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-right"
        />
      </svg>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;

exports[`Should render component at page 1 1`] = `
<ul
  className="pagination pull-right"
>
  <li
    className="page-item disabled"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#0"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-left"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-left"
        />
      </svg>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
  </li>
  <li
    className="page-item active"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#2"
    >
      2
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#3"
    >
      3
    </a>
  </li>
  <li
    className="dropdown page-item"
  >
    <a
      className="dropdown-toggle page-link"
      data-toggle="dropdown"
      href="javascript:;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
      <span
        className="sr-only"
      />
    </a>
    <div
      className="dropdown-menu dropdown-menu-top-center"
    >
      <ul
        className="inline-scroller link-list"
        data-current-index="4"
      >
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#4"
          >
            <span
              className="sr-only"
            />
            4
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#5"
          >
            <span
              className="sr-only"
            />
            5
          </a>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#6"
    >
      6
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#2"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-right"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-right"
        />
      </svg>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;

exports[`Should render component at page 2 1`] = `
<ul
  className="pagination pull-right"
>
  <li
    className="page-item disabled"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#-1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-left"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-left"
        />
      </svg>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="page-item active"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#0"
    >
      0
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="dropdown page-item"
  >
    <a
      className="dropdown-toggle page-link"
      data-toggle="dropdown"
      href="javascript:;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
      <span
        className="sr-only"
      />
    </a>
    <div
      className="dropdown-menu dropdown-menu-top-center"
    >
      <ul
        className="inline-scroller link-list"
        data-current-index="2"
      >
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#2"
          >
            <span
              className="sr-only"
            />
            2
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#3"
          >
            <span
              className="sr-only"
            />
            3
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#4"
          >
            <span
              className="sr-only"
            />
            4
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#5"
          >
            <span
              className="sr-only"
            />
            5
          </a>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#6"
    >
      6
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-right"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-right"
        />
      </svg>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;

exports[`Should render component at page 3 1`] = `
<ul
  className="pagination pull-right"
>
  <li
    className="page-item disabled"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#-1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-left"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-left"
        />
      </svg>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="page-item active"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#0"
    >
      0
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="dropdown page-item"
  >
    <a
      className="dropdown-toggle page-link"
      data-toggle="dropdown"
      href="javascript:;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
      <span
        className="sr-only"
      />
    </a>
    <div
      className="dropdown-menu dropdown-menu-top-center"
    >
      <ul
        className="inline-scroller link-list"
        data-current-index="2"
      >
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#2"
          >
            <span
              className="sr-only"
            />
            2
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#3"
          >
            <span
              className="sr-only"
            />
            3
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#4"
          >
            <span
              className="sr-only"
            />
            4
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#5"
          >
            <span
              className="sr-only"
            />
            5
          </a>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#6"
    >
      6
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-right"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-right"
        />
      </svg>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;

exports[`Should render component at page 4 1`] = `
<ul
  className="pagination pull-right"
>
  <li
    className="page-item disabled"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#-1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-left"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-left"
        />
      </svg>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="page-item active"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#0"
    >
      0
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="dropdown page-item"
  >
    <a
      className="dropdown-toggle page-link"
      data-toggle="dropdown"
      href="javascript:;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
      <span
        className="sr-only"
      />
    </a>
    <div
      className="dropdown-menu dropdown-menu-top-center"
    >
      <ul
        className="inline-scroller link-list"
        data-current-index="2"
      >
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#2"
          >
            <span
              className="sr-only"
            />
            2
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#3"
          >
            <span
              className="sr-only"
            />
            3
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#4"
          >
            <span
              className="sr-only"
            />
            4
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#5"
          >
            <span
              className="sr-only"
            />
            5
          </a>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#6"
    >
      6
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-right"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-right"
        />
      </svg>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;

exports[`Should render component at page 5 1`] = `
<ul
  className="pagination pull-right"
>
  <li
    className="page-item disabled"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#-1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-left"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-left"
        />
      </svg>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="page-item active"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#0"
    >
      0
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#1"
    >
      1
    </a>
  </li>
  <li
    className="dropdown page-item"
  >
    <a
      className="dropdown-toggle page-link"
      data-toggle="dropdown"
      href="javascript:;"
    >
      <span
        aria-hidden="true"
      >
        ...
      </span>
      <span
        className="sr-only"
      />
    </a>
    <div
      className="dropdown-menu dropdown-menu-top-center"
    >
      <ul
        className="inline-scroller link-list"
        data-current-index="2"
      >
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#2"
          >
            <span
              className="sr-only"
            />
            2
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#3"
          >
            <span
              className="sr-only"
            />
            3
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#4"
          >
            <span
              className="sr-only"
            />
            4
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
            href="#5"
          >
            <span
              className="sr-only"
            />
            5
          </a>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      href="#6"
    >
      6
    </a>
  </li>
  <li
    className="page-item"
    onClick={[Function]}
  >
    <a
      className="page-link"
      data-senna-off={true}
      href="#1"
      role="button"
    >
      <svg
        className="lexicon-icon lexicon-icon-angle-right"
        focusable="false"
        role="presentation"
      >
        <use
          href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#angle-right"
        />
      </svg>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;
